service/worker/processor.go

```diff
@@ -65,6 +94,12 @@ type (
 		metricsClient    metrics.Client
 		domainReplicator DomainReplicator
 		historyClient    history.Client
+
+		sync.Mutex
+		// worker in retry is used by underlying processor when doing retry on a task
+		// this help the replicator / underlying processor understanding the overall
+		// situation and giveup retrying
+		workerInRetry map[int]struct{}
```

Comment:
Can we use atomic increment/decrement for keeping the retry count?  Then we don't need  lock

Reply:
done